### DISCUSSION TITLE
Support Router Matching Syntax

### DISCUSSION THREAD

**Discussion by Nirlah** [nuxt/nuxt#18123] [Ideas]
It could be great for pages to support the [Router’s Matching Syntax](https://next.router.vuejs.org/guide/essentials/route-matching-syntax.html).

For example, a page/dir with a number filter `foo-[bar(\d+)].vue`

**Comment by viii47** [comment]
This won't work because certain characters are not allowed to be used on file generation. If you want to use RegEx-Patterns for routes then you have to create a general routing definition file (like in the link you've posted) with matching orders (e.g. /foo-[barnumbersonly] to be /foo-[bar(\d+)])

**Comment by Nirlah** [comment]
Hmm, it makes sense... I wonder if there could be another way to allow such a feature validly.
Also, is there a way to config the routes outside the Routes directory?

**Reply by levchak0910** [reply]
@Nirlah for expample
```
defineNuxtConfig({
  hooks: {
    "pages:extend"(pages) {
      pages.push({
          name: "shop",
          path: "/:shop(.+-shop)/",
          strict: true,
          file: "@/path-to-Shop.page.vue",
        })
    }
  }
})
```

**Reply by Nirlah** [reply]
Perfect! Thanks :)
